Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/TelemetryBufferTest.cs

```diff
@@ -112,6 +116,31 @@ public void AddsTelemetryToBufferUntilItReachesMax()
             }
 
             [TestMethod]
+            public void TelemetryBufferDoNotGrowBeyondMaxBacklogSize()
```

Comment:
TelemetryBuffer.cs is a class which is present in InMemoryChannel and in ServerTelemetryChannel - they share the name but are independent class and files. And they are tested separately as well.